src/utilities.js

```diff
@@ -69,68 +53,79 @@ const filterObjectsFromSave = function (save, list) {
- * @returns {string|*} valid result
+ * Returns if object is number or not
+ * @param {string}n String to be checked
+ * @returns {boolean}
  */
-const getValidData = function (type, min, max, selected, previous) {
-    switch (type) {
-        case "number":
-            return validateNumber(parseFloat(selected), min, max, previous);
-        case "text":
-            if (selected) {
-                return selected.toString();
-            } else if (previous) {
-                return previous.toString();
-            } else {
-                return "";
-            }
-        default:
-            return selected;
-    }
+const isNumeric = function (n) {
+    return !isNaN(parseFloat(n)) && isFinite(n);
 };
 
-module.exports.getValidData = getValidData;
+module.exports.isNumeric = isNumeric;
 
 /**
  * Returns a valid number within constrains
  * @param selected value to be checked
  * @param min minimum value allowed for numbers
  * @param max maximum value allowed for numbers
- * @param previous last valid entry
  * @returns {number|float} valid number
  */
-const validateNumber = function (selected, min, max, previous) {
-    if (isNaN(selected)) {
-        if (previous !== undefined) {
-            if (!isNaN(parseFloat(previous))) {
-                return validateNumber(parseFloat(previous), min, max, undefined);
-            } else {
-                return 0;
-            }
+const clampNumber = function (selected, min, max) {
+    if (selected < min) {
+        return min;
+    } else if (selected > max) {
+        return max;
+    } else {
+        return selected;
+    }
+};
+
+module.exports.clampNumber = clampNumber;
+
+
+/**
+ * Validator for number fields
+ * @param {*}num value to be checked
+ * @param {number}min minimum value allowed for numbers
+ * @param {number}max maximum value allowed for numbers
+ * @param {number|string}defaultValue default value to be passed
+ * @returns {number} valid number
+ */
+const getValidNumber = function (num, min, max, defaultValue = 0) {
+    if (!isNumeric(num)) {
+        if (!isNumeric(defaultValue)) {
+            num = 0;
         } else {
-            if (min !== undefined) {
-                return min;
-            } else {
-                return 0;
-            }
+            return getValidNumber(defaultValue, min, max);
         }
-    } else {
-        if (selected < min) {
-            return min;
-        } else if (selected > max) {
-            return max;
+    } else if (typeof num !== "number") {
+        num = parseFloat(num);
+    }
+    return clampNumber(num, min, max);
+};
+
+module.exports.getValidNumber = getValidNumber;
+
+/**
+ * Returns a valid text
+ * @param val Value to be check
+ * @param defaultValue Default value to be passed
+ * @returns {*} Valid String
+ */
+const getValidText = function (val, defaultValue = "") {
+    if (val === undefined) {
+        if (defaultValue === undefined) {
+            val = defaultValue;
```

Comment:
I don't get here, `val = undefined` ?

Reply:
if (string) will produce false as a result if string is empty. Empty string can be considered valid text henceforth I check if string is defined or not

Reply:
Oh I replaced that code block back then. I am comparing now... It should be fixed

Reply:
ok I wonder that could not find in a head, it old outdated commit